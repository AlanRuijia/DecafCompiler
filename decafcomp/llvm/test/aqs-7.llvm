; ModuleID = 'QuickSort'

define void @quickSort(i32, i32) {
entry:
  %left = alloca i32
  %right = alloca i32
  store i32 %0, i32* %left
  store i32 %1, i32* %right
  %right1 = load i32, i32* %right
  %left2 = load i32, i32* %left
  %SubTmp = sub i32 %right1, %left2
  %LEQTmp = icmp sle i32 %SubTmp, 0
  %IfCond = icmp eq i1 %LEQTmp, true
  br i1 %IfCond, label %Then, label %Else

Then:                                             ; preds = %entry
  ret void
  br label %IfCont
  ret void
}
