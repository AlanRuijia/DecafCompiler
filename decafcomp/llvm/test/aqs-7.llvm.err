; ModuleID = 'QuickSort'

define void @quickSort(i32, i32) {
entry:
  %left = alloca i32
  %right = alloca i32
  store i32 %0, i32* %left
  store i32 %1, i32* %right
  ret void
}
