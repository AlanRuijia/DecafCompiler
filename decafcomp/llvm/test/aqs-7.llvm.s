	.section	__TEXT,__text,regular,pure_instructions
	.macosx_version_min 10, 12
	.globl	_quickSort
	.align	4, 0x90
_quickSort:                             ## @quickSort
	.cfi_startproc
## BB#0:                                ## %entry
	movl	%edi, -4(%rsp)
	movl	%esi, -8(%rsp)
	retq
	.cfi_endproc


.subsections_via_symbols
