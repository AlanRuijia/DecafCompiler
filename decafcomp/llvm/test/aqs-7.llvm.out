Inserting method decl into symbol table...
Inserting block into symbol table...
Generating program...
Generating package...

Symbol Table: left right 
Symbol Table: quickSort 
Symbol Table: 
Debug message: Generating MethodDecl...
In IDTypeStringSpecialAST
Allocating left
Serching table...
FetchedObject: Found left
In IDTypeStringSpecialAST
Allocating right
Serching table...
FetchedObject: Found right
Serching table...
FetchedObject: Found left
Serching table...
FetchedObject: Found right
Method Block CodeGen...
Block CodeGen......
Serching table...
FetchedObject: Found right
Serching table...
FetchedObject: Found left
OpId 1
Starting NumberExpr Codegen
Finished NumberExpr Codegen
OpId 9
